### ISSUE TITLE
There is a lack of Ethernet examples

### ISSUE THREAD

**Issue by GeorgeFlorian** [espressif/arduino-esp32#3030]
OS: Linux Mint 19.1
Board: Olimex ESP32-EVB

Hello !

I am trying to implement an Ethernet connection instead of an WiFi connection.

This is how I start a WiFi connection in STA Mode right now:
```
       if(!WiFi.config(local_IP_STA, gateway_STA, subnet_STA, local_IP_STA, primaryDNS)) {
         logOutput((String)"Couldn't configure STATIC IP ! Starting DHCP IP !");
       }
       delay(50);
       WiFi.begin(x[0].c_str(),x[1].c_str());
```
What is the equivalent for the above code when using Ethernet ?

I also have some other questions:
 * Can WiFi and Ethernet be used in the same time and how would this work. Will the ESP32 be able to handle 2 different types of connections in the same time or you can code-in both of these but only use one at a time ?
 * Can Ethernet be used in AP Mode ?
 * How come the ETH example is in WiFi examples ? Doesn't ETH deserve a separate directory with its own examples ?

Thank you !

**Comment by atanisoft** [comment]
I'm guessing you missed these: https://github.com/espressif/arduino-esp32/tree/master/libraries/WiFi/examples

The ETH_XXX ones are for Ethernet, they are present in the WiFi library as that library provides the necessary infrastructure.

**Comment by GeorgeFlorian** [comment]
> I'm guessing you missed these: https://github.com/espressif/arduino-esp32/tree/master/libraries/WiFi/examples
> 
> The ETH_XXX ones are for Ethernet, they are present in the WiFi library as that library provides the necessary infrastructure.

Of course I've missed them then ! ETH is not WiFi, is it ?
I've managed to make it work since I've posted the question.

**Comment by atanisoft** [comment]
On the ESP32 platform the ETH and WiFi systems share the same TCP stack and underlying PHY. So they are in a way coupled.

If your issue has been solved please consider closing this issue :)

**Comment by GeorgeFlorian** [comment]
> On the ESP32 platform the ETH and WiFi systems share the same TCP stack and underlying PHY. So they are in a way coupled.
> 
> If your issue has been solved please consider closing this issue :)

How is anyone supposed to know that ?

**Comment by atanisoft** [comment]
> How is anyone supposed to know that ?

Good question. Most likely via the IDF docs but those are a bit terse or hard to follow in some areas.
